cudax/test/common/testing.cuh

```diff
@@ -44,14 +45,24 @@ inline void __device__ cudax_require_impl(
   }
 }
 
-// TODO make it work on NVC++
-#ifdef __CUDA_ARCH__
-#  define CUDAX_REQUIRE(condition) cudax_require_impl(condition, #condition, __FILE__, __LINE__, __PRETTY_FUNCTION__);
-#else
-#  define CUDAX_REQUIRE REQUIRE
-#endif
+#define CUDAX_REQUIRE(condition)                                                                           \
+  NV_IF_ELSE_TARGET(NV_IS_DEVICE,                                                                          \
+                    (cudax_require_impl(condition, #condition, __FILE__, __LINE__, __PRETTY_FUNCTION__);), \
+                    (REQUIRE(condition);))
 
-bool constexpr __host__ __device__ operator==(const dim3& lhs, const dim3& rhs)
+#define CUDAX_CHECK(condition)                                                                             \
```

Comment:
Note: I guess CUDAX_CHECK behaves as CUDAX_REQUIRE in device code, but I don't think its a problem and I don't see any other way to do it